src/main/scala/org/opennetworkinsight/dns/DNSWordCreation.scala

```diff
@@ -1,239 +1,95 @@
-        // don't use: country_codes.exists(spliturl(numparts-1).contains) this doesn't test exact match, might just match substring
-        if (Array(spliturl(numparts - 1)).exists(country_codes.value contains _)) {
-          domain = spliturl(numparts - 3)
-          if (1 <= numparts - 3) {
-            subdomain = spliturl.slice(0, numparts - 3).mkString(".")
-          }
-        }
-        else {
-          domain = spliturl(numparts - 2)
-          subdomain = spliturl.slice(0, numparts - 2).mkString(".")
-        }
-      }
-      Array(domain, subdomain, {
-        if (subdomain != "None") {
-          subdomain.length.toDouble
-        } else {
-          0.0
-        }
-      }, numparts.toDouble)
-    }
-
-    def udfStringEntropy() = udf((subdomain: String) => Entropy.stringEntropy(subdomain))
-
-    def udfWordCreation(frameLengthCuts: Array[Double],
-                        timeCuts: Array[Double],
-                        subdomainLengthCuts: Array[Double],
-                        entropyCuts: Array[Double],
-                        numberPeriodsCuts: Array[Double]) =
-      udf((topDomain: String,
-           frameLength: Int,
-           unixTimeStamp: Long,
-           subdomainLength: Double,
-           subdomainEntropy: Double,
-           numberPeriods: Double,
-           dnsQueryType: Int,
-           dnsQueryRcode: Int) => dnsWord(topDomain, frameLength, unixTimeStamp, subdomainLength, subdomainEntropy,
-        numberPeriods, dnsQueryType, dnsQueryRcode, frameLengthCuts, timeCuts, subdomainLengthCuts, entropyCuts, numberPeriodsCuts))
-
-    def dnsWord(topDomain: String,
-                frameLength: Int,
-                unixTimeStamp: Long,
-                subdomainLength: Double,
-                subdomainEntropy: Double,
-                numberPeriods: Double,
-                dnsQueryType: Int,
-                dnsQueryRcode: Int,
-                frameLengthCuts: Array[Double],
-                timeCuts: Array[Double],
-                subdomainLengthCuts: Array[Double],
-                entropyCuts: Array[Double],
-                numberPeriodsCuts: Array[Double]): String = {
-      Seq(topDomain ,
-        Quantiles.bin(frameLength.toDouble, frameLengthCuts) ,
-        Quantiles.bin(unixTimeStamp.toDouble, timeCuts) ,
-        Quantiles.bin(subdomainLength, subdomainLengthCuts) ,
-        Quantiles.bin(subdomainEntropy, entropyCuts) ,
-        Quantiles.bin(numberPeriods, numberPeriodsCuts) ,
-        dnsQueryType ,
-        dnsQueryRcode).mkString("_")
-    }
+import org.opennetworkinsight.utilities.DomainProcessor.{DomainInfo, extractDomainInfo}
+import org.opennetworkinsight.utilities.Quantiles
+
+
+/**
+  * Convert DNS log entries into "words" for topic modelling analyses.
+  *
+  * @param frameLengthCuts
+  * @param timeCuts
+  * @param subdomainLengthCuts
+  * @param entropyCuts
+  * @param numberPeriodsCuts
+  * @param topDomainsBC
+  */
+class DNSWordCreation(frameLengthCuts: Array[Double],
```

Comment:
do these parameters look familiar?

they are basically exactly the state in a DNSSuspiciousConnectsModel....

if it weren't for the OA side information requiring the "words" we could just make this a method in the DNSSuspiciousConnectsModel  (or a closely tied Serializable class)